### PR TITLE
Pwd strength checker in csharp

### PR DESCRIPTION
**Pull request by tbhaxor**

### CODE REVIEW

File: code/data_structures/heap/min_heap/min_heap.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+//gives the parent value.
+int parent(int i){
+	int p;
+	if(i%3==0){p=i/3-1;}
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: Don't use braces here, it seems redundant (unless you separate the lines).

File: code/data_structures/heap/min_heap/min_heap.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+//gives the parent value.
+int parent(int i){
+	int p;
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: If you want, you could eliminate the variable, and directly return the result.
I would simplify this function to:

```C
return (i%3==) ? (i/3-1) : (i/3);
```

File: code/data_structures/heap/min_heap/min_heap.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	if(i%3==0){p=i/3-1;}
+	else{p=i/3;}
+	return p;
+}
+
+void choices()
+{
+	printf("\n\nYou have following chices.\n");
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: You have _the_ following ch _o_ ices.

File: code/data_structures/heap/min_heap/min_heap.c [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	printf("i or I: \tTo insert a new element into the heap.\n");
+	printf("d or D: \tTo delete the minimum element from the list.\n");
+	printf("q or Q: \tTo quit.\n\n");
+}
+
+//minheap function. Building heap ne by one.
+void minheapify(int a[],int size){
+	for(int i=1;i<size;i++){
+		int s=i;
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: Why a new variable, just use i?

File: code/data_structures/heap/min_heap/min_heap.c [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+			s=p;
+			p=parent(s);
+		}
+		
+	}
+}
+
+//insert function to insert in the heap.
+int insert(int a[],int x,int size){
+	size++;			//increase the size of the heap by one and insert the element at the last position
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: Pre increment the variable.

File: code/data_structures/heap/min_heap/min_heap.c [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+int deletemin(int a[],int size){
+	
+	//check if size is zero
+	if(size==0){printf("can not delete from empty heap\n");}
+
+	else{
+		a[0]=a[size-1];	//the first element is replaced by the last and decrease the size by one.
+		size--;
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: Pre decrement.

File: code/data_structures/heap/min_heap/min_heap.c [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+		minheapify(a,size); 	//apply minheapify
+		return size;
+	}
+}
+
+
+//to print the heap.
+void showheap(int a[],int size){
+	printf("\n\nThe heap is.\n");
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: . -> :

File: code/data_structures/heap/min_heap/min_heap.c [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+//main starts here
+int main(){
+	int n;
+	int x;
+	char c;
+
+	printf("\nEnter the no. of elements in the heap.\n");
+	scanf("%d",&n);
+
+	int a[1000];
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: While your approach is fine, this limits the size of the heap to 1000 numbers, so I suggest you use malloc after the use enters the size, or VLAs.

File: code/data_structures/heap/min_heap/min_heap.c [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+	
+	//showheap after insertion
+	showheap(a,size);
+	choices();
+	printf("Enter your choice.\n");
+	scanf("%c",&c);
+	scanf("%c",&c);	
+	
+	//exit condition is that c=q or Q
+	while(c!='q' || c!='Q'){
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: toupper/tolower would help here, whenever reading a choice.

File: code/data_structures/heap/min_heap/min_heap.c [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+				printf("%d is deleted\n",a[0]);
+				size=deletemin(a,size);
+				showheap(a,size);
+			}
+		}
+		printf("Enter your choice.\n");
+		scanf("%c",&c);
+		scanf("%c",&c);
+
+		if(c=='q' || c=='Q') break;
```
Comment (arnavb) [inline code/data_structures/heap/min_heap/min_heap.c]: Not needed.

File: code/mathematical-algorithms/armstrong_numbers/armstrong_number.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  $sum = 0;
+  $dupli = $number;
+  while($dupli != 0)
+  {
+    $digit = $dupli % 10;
+    $sum += $digit * $digit * $digit;
+    $dupli/=10;
+  }
+  if ($sum == $number)
+    return true;
```
Comment (arnavb) [inline code/mathematical-algorithms/armstrong_numbers/armstrong_number.php]: Simplify to `return $sum == $number;`

File: code/unclassified/biggest_Of_N_Numbers/biggest_of_n_numbers.cpp [hunk 12]
```diff
@@ -0,0 +1,25 @@
+// compile with c++11 standard
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+using namespace std;
+
+int biggest_of_n(vector<int> v)
```
Comment (arnavb) [inline code/unclassified/biggest_Of_N_Numbers/biggest_of_n_numbers.cpp]: No need for a separate function.

File: code/string-algorithms/password_strength_checker/pw_checker.cs [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+{
+    class Program
+    {
+        static bool chk_strength(string password)
+        {
+            bool special_char = password.Any(ch => !char.IsLetterOrDigit(ch));
+            bool digit = password.Any(char.IsDigit);
+            bool uppercase = password.Any(char.IsUpper);
+            
+            if (password.Length < 8)
```
Comment (arnavb) [inline code/string-algorithms/password_strength_checker/pw_checker.cs]: You could use an or statement to combine the conditionals.

File: code/string-algorithms/lipogram_checker/lipogram_checker.cs [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            int count = 0;
+            string alphabets = "abcdefghijklmnopqrstuvwxyz";
+            str = str.ToLower();
+            for (int i = 0; i < alphabets.Length; i++)
+            {
+                int index = str.IndexOf(alphabets[i]);
+
+                if (index < 0 || index > str.Length)
+                    count ++;
```
Comment (arnavb) [inline code/string-algorithms/lipogram_checker/lipogram_checker.cs]: Remove space before ++.

**Comment by tbhaxor** [comment]
@arnavb  i am pushing all these fixes in new branch

File: code/string-algorithms/lipogram_checker/lipogram_checker.cs (hunk 14 above)
Comment (tbhaxor) [reply code/string-algorithms/lipogram_checker/lipogram_checker.cs]: kindly review my new commit [here](https://github.com/OpenGenus/cosmos/pull/3000)

File: code/unclassified/biggest_Of_N_Numbers/biggest_of_n_numbers.cpp (hunk 12 above)
Comment (tbhaxor) [reply code/unclassified/biggest_Of_N_Numbers/biggest_of_n_numbers.cpp]: @arnavb plz review my new commit [here](https://github.com/OpenGenus/cosmos/pull/2990)

File: code/mathematical-algorithms/armstrong_numbers/armstrong_number.php (hunk 11 above)
Comment (tbhaxor) [reply code/mathematical-algorithms/armstrong_numbers/armstrong_number.php]: @arnavb plz review my new commit [here](https://github.com/OpenGenus/cosmos/pull/2995)

Comment (arnavb) [reply code/mathematical-algorithms/armstrong_numbers/armstrong_number.php]: What about the rest of the changes?

**Comment by arnavb** [comment]
You need to switch your branches, since all your changes are everywhere.

**Pull request closed without merging**